### PR TITLE
adding abort on error to reset state machine

### PR DESCRIPTION
**Pull request by marketideas**
Errors on I2C bus can leave the I2C state machine in a non-idle state.  Calling the MCUX I2C_MasterTransferAbort function in the FSL library will properly reset this state so that future transactions will work.

Also, commented out some lines in pinux.c for the FRDM-K64F board. These lines were set for Ethernet, but are not used by the K64F board. These lines prevented UART 3 from working correctly (or at all)

### CODE REVIEW

**Comment by carlescufi** [comment]
@marketideas thanks for the PR. Please note that there are no merge commits allowed in Pull Requests, and that you need to fix your commit message formatting according to:
https://docs.zephyrproject.org/latest/contribute/index.html#contribution-workflow

**Comment by henrikbrixandersen** [comment]
@marketideas Thank you for the PR. I believe you have included a bit too many unrelated changes and commits in this PR. Please clean it up to only contain the exact changes needed for the bug fix (look at the "Commits" and "Files changed" tabs).

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
@marketideas Thank you for the PR. I believe you have included a bit too many unrelated changes and commits in this PR. Please clean it up to only contain the exact changes needed for the bug fix (look at the "Commits" and "Files changed" tabs).

File: drivers/i2c/i2c_mcux.c [hunk 1]
```diff
@@ -150,6 +152,7 @@ static int i2c_mcux_transfer(struct device *dev, struct i2c_msg *msgs,
 		 * successfully. e.g., nak, timeout, lost arbitration
 		 */
 		if (data->callback_status != kStatus_Success) {
+			I2C_MasterTransferAbort(base, &data->handle);
```
Comment (henrikbrixandersen) [inline drivers/i2c/i2c_mcux.c:155]: ... and this one. Please revisit your patch and reduce it to only include the actual bug fix.

File: drivers/i2c/i2c_mcux.c [hunk 2]
```diff
@@ -134,12 +135,13 @@ static int i2c_mcux_transfer(struct device *dev, struct i2c_msg *msgs,
...
 		/* Start the transfer */
 		status = I2C_MasterTransferNonBlocking(base,
-				&data->handle, &transfer);
+						       &data->handle, &transfer);
 
 		/* Return an error if the transfer didn't start successfully
 		 * e.g., if the bus was busy
 		 */
 		if (status != kStatus_Success) {
+			I2C_MasterTransferAbort(base, &data->handle);
```
Comment (henrikbrixandersen) [inline drivers/i2c/i2c_mcux.c:144]: As far as I can tell, there should only be to lines added for this PR, namely this one...

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

**Comment by henrikbrixandersen** [comment]
I have created an alternative PR #26761. Please either fix this one or review that one.

**Comment by henrikbrixandersen** [comment]
Superseeded by #26761. Please open a new issue/PR for the FRDM-K64F UART3 issue as needed.

**Pull request closed without merging**